pyro/distributions/multivariate_cauchy.py

```diff
@@ -0,0 +1,51 @@
+import torch
+from torch.distributions import constraints
+from torch.distributions.transforms import AffineTransform
+from torch.distributions.cauchy import Cauchy
+from pyro.distributions.torch import TransformedDistribution
+from pyro.distributions.torch_distribution import TorchDistributionMixin
+
+
+class _Cauchy(Cauchy, TorchDistributionMixin):
+    pass
+
+
+class MultivariateCauchy(TransformedDistribution):
+    r"""
+    Creates a multivariate Cauchy distribution parameterized by
+    :attr:`loc` and :attr:`scale_tril` where::
+
+        X ~ Cauchy(loc, 1.0)
+        Y = scale_tril @ X ~ MultivariateCauchy(loc, scale_tril)
+
+    :param torch.tensor loc: the location parameter of the distribution, a D-dimensional vector.
+    :param torch.tensor scale_tril: the scale parameter of the distribution, a D x D lower-triangular matrix.
+    """
+    arg_constraints = {'loc': constraints.real, 'scale_tril': constraints.lower_triangular}
+    support = constraints.real
+    has_rsample = True
+
+    def __init__(self, loc, scale_tril, validate_args=None):
+        base_dist = _Cauchy(loc, 1.0).to_event(1)
+        transforms = [AffineTransform(torch.zeros_like(loc), scale_tril, event_dim=1)]
```

Comment:
oh wait this won't work affine is only for pointwise stuff

Reply:
Hmm..yeah, this won't work. I think you can probably easily adapt @fehiepsi's [MultivariateAffineTransform](https://github.com/pyro-ppl/numpyro/blob/033393afc92563c82ef9723907f36b8774065b9b/numpyro/distributions/transforms.py#L336) for this purpose, and we can later move it upstream to pytorch if it seems useful.

Reply:
Yeah, we should add that to `pyro.distributions.transforms` and later move it upstream.

Reply:
done